airflow/models/connection.py

```diff
@@ -186,16 +186,43 @@ def _normalize_conn_type(conn_type):
             conn_type = conn_type.replace("-", "_")
         return conn_type
 
+    @staticmethod
+    def split_uri_to_parts(uri):
+        c = uri.count("://")
```

Comment:
Is it ever possible for this to return >=2 if the escape syntax is used…? A valid URL can never contain more than one `://` sequence.

Reply:
Yeah makes sense. Also could be a good idea to add something in the Connection form to emit an error if the user puts in an invalud URL. Probably cleaner to do in a separate PR though.

Reply:
@uranusjr do you mean on the UI? I am not sure if we have a check for that, it would be great to do that though.

Reply:
I think this can be done in `ConnectionForm` (see WTForms documentation to add validators). We should probably do that on the `host` field.

Reply:
Nice. Let me come up with a separate issue and PR for that :)